src/Avalonia.Vulkan/VulkanPlatformInterface.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using Avalonia.Vulkan.Surfaces;
+using Silk.NET.Vulkan;
+
+namespace Avalonia.Vulkan
+{
+    public class VulkanPlatformInterface : IDisposable
+    {
+        private static VulkanOptions _options;
+
+        private VulkanPlatformInterface(VulkanInstance instance)
+        {
+            Instance = instance;
+            Api = instance.Api;
+        }
+
+        public VulkanPhysicalDevice PhysicalDevice { get; private set; }
+        public VulkanInstance Instance { get; }
+        public VulkanDevice Device { get; private set; }
+        public Vk Api { get; private set; }
+
+        public void Dispose()
+        {
+            Device?.Dispose();
+            Instance?.Dispose();
+            Api?.Dispose();
+        }
+
+        private static VulkanPlatformInterface TryCreate()
+        {
+            _options = AvaloniaLocator.Current.GetService<VulkanOptions>() ?? new VulkanOptions();
+
+#if NET6_0_OR_GREATER
+            if (OperatingSystem.IsAndroid())
+                Silk.NET.Core.Loader.SearchPathContainer.Platform = Silk.NET.Core.Loader.UnderlyingPlatform.Android;
+#endif
+
+            var instance = VulkanInstance.Create(_options);
```

Comment:
This can throw when instance creation fails and instead of returning null it will crash the entire application.